document/core/appendix/embedding.rst

```diff
@@ -0,0 +1,526 @@
+
+* If :math:`\X{ti}.\TIMAX` is non-empty and smaller than :math:`n` added to the length of :math:`\X{ti}.\TIELEM`, then return :math:`\ERROR`.
+
+* Append :math:`n` empty elements to :math:`\X{ti}.\TIELEM`.
+
+* Return the updated store.
+
+.. math::
+   \begin{array}{lclll}
+   \F{grow\_table}(S, a, n) &=& S' && (
+     \begin{array}[t]{@{}r@{~}l@{}}
+     \iff & \X{ti} = S.\STABLES[a] \\
+     \wedge & (\X{ti}.\TIMAX = \epsilon \vee |\X{ti}.\TIELEM| + n \leq \X{ti}.\TIMAX) \\
+     \wedge & S' = S \with \STABLES[a].\TIELEM = \X{ti}.\TIELEM~(\epsilon)^n) \\
+     \end{array} \\
+   \F{grow\_table}(S, a, n) &=& \ERROR && (\otherwise) \\
+   \end{array}
+
+
+.. index:: memory, memory address, store, memory instance, memory type, byte
+.. _embed-mem:
+
+Memories
+~~~~~~~~
+
+.. _embed-alloc-mem:
+
+:math:`\F{alloc\_mem}(\store, \memtype) : (\store, \memaddr)`
+................................................................
+
+* Let :math:`\memaddr` be the result of :ref:`allocating a memory <alloc-mem>` in :math:`\store` with :ref:`memory type <syntax-memtype>` :math:`\memtype`.
+
+* Return the new store paired with :math:`\memaddr`.
+
+.. math::
+   \begin{array}{lclll}
+   \F{alloc\_mem}(S, \X{mt}) &=& (S', \X{a}) && (\iff \allocmem(S, \X{mt}) = S', \X{a}) \\
+   \end{array}
+
+
+.. _embed-read-mem:
+
+:math:`\F{read\_mem}(\store, \memaddr, i) : \byte ~|~ \error`
+.............................................................
+
+* Assert: :math:`\store.\SMEMS[\memaddr]` exists.
+
+* Assert: :math:`i` is a non-negative integer.
+
+* Let :math:`\X{mi}` be the :ref:`memory instance <syntax-meminst>` :math:`\store.\SMEMS[\memaddr]`.
+
+* If :math:`i` is larger than or equal to the length if :math:`\X{mi}.\MIDATA`, then return :math:`\ERROR`.
+
+* Else, return the  :ref:`byte <syntax-byte>` :math:`\X{mi}.\MIDATA[i]`.
+
+.. math::
+   \begin{array}{lclll}
+   \F{read\_mem}(S, a, i) &=& b && (\iff S.\SMEMS[a].\MIDATA[i] = b) \\
+   \F{read\_mem}(S, a, i) &=& \ERROR && (\otherwise) \\
+   \end{array}
+
+
+.. _embed-write-mem:
+
+:math:`\F{write\_mem}(\store, \memaddr, i, \byte) : \store ~|~ \error`
+......................................................................
+
+* Assert: :math:`\store.\SMEMS[\memaddr]` exists.
+
+* Assert: :math:`i` is a non-negative integer.
+
+* Let :math:`\X{mi}` be the :ref:`memory instance <syntax-meminst>` :math:`\store.\SMEMS[\memaddr]`.
+
+* If :math:`i` is larger than or equal to the length if :math:`\X{mi}.\MIDATA`, then return :math:`\ERROR`.
+
+* Replace :math:`\X{mi}.\MIDATA[i]` with :math:`\byte`.
+
+* Return the updated store.
+
+.. math::
+   \begin{array}{lclll}
+   \F{write\_mem}(S, a, i, b) &=& S' && (\iff S' = S \with \SMEMS[a].\MIDATA[i] = b) \\
+   \F{write\_mem}(S, a, i, b) &=& \ERROR && (\otherwise) \\
+   \end{array}
+
+
+.. _embed-grow-mem:
+
+:math:`\F{grow\_mem}(\store, \memaddr, n) : \store ~|~ \error`
```

Comment:
Is it possible/worthwhile to share this somehow with the [grow_memory instruction](https://webassembly.github.io/spec/exec/instructions.html#exec-grow-memory)?

Also, according to the note that instruction can fail for unspecified reasons. Should this function have similar non-determinism?